arch/x86/kvm/nitro_x86.c

```diff
@@ -107,9 +107,53 @@ void nitro_wait(struct kvm_vcpu *vcpu){
   return;
 }
 
-void nitro_report_event(struct kvm_vcpu *vcpu){
-  nitro_wait(vcpu);
-  vcpu->nitro.event.present = false;
+void nitro_report_event(struct kvm_vcpu *vcpu, uint64_t syscall_nb){
+	struct kvm* kvm = vcpu->kvm;
+
+	// if no filter, report all events
+	// or if there is a filter
+	if (kvm->nitro.syscall_filter_size == 0
+			|| nitro_find_syscall(kvm, syscall_nb) == true)
+	{
+		nitro_wait(vcpu);
+		vcpu->nitro.event.present = false;
+	}
+}
+
+void nitro_process_event(struct kvm_vcpu *vcpu)
+{
+	uint64_t syscall_nb = 0;
+	if (vcpu->nitro.event.direction == ENTER)
+	{
+		syscall_nb = vcpu->nitro.event.regs.rax;
+		// create new syscall stack item
+		struct syscall_stack_item *item = kmalloc(sizeof(struct syscall_stack_item), GFP_KERNEL);
```

Comment:
I guess we will leak kernel memory if VMs are abruptly interrupted here. 
The list needs to be eventually cleared at VM cleanup.